packages/runtime/src/runtimes/react/controls/rich-text.tsx

```diff
@@ -0,0 +1,206 @@
+import {
+  ReactNode,
+  useCallback,
+  useEffect,
+  useRef,
+  useState,
+  KeyboardEvent as ReactKeyboardEvent,
+  FocusEvent as ReactFocusEvent,
+} from 'react'
+
+import {
+  RichTextControl,
+  RichTextControlData,
+  RichTextDefinition,
+} from '../../../controls/rich-text'
+import { Text } from '../../../components'
+import { RichTextEditor } from '../../../components/builtin/Text/components/RichTextEditor'
+// @ts-expect-error: no types for 'slate-hotkeys'
+import Hotkeys from 'slate-hotkeys'
+import { isHotkey } from 'is-hotkey'
+
+import { Value } from 'slate'
+import { Editor, OnChangeParam } from 'slate-react'
+import { useIsInBuilder } from '../../../react'
+import { cx } from '@emotion/css'
+import { pollBoxModel } from './slot'
+export type RichTextControlValue = ReactNode
+
+const COMMIT_DEBOUNCE_DELAY = 500
+
+export function useRichText(
+  data: RichTextControlData,
+  control: RichTextControl | null,
+  descriptor: RichTextDefinition,
+) {
+  const [element, setElement] = useState<HTMLElement | null>(null)
+  const [editor, setEditor] = useState<Editor | null>(null)
+
+  useEffect(() => {
+    if (element == null || control == null) return
+
+    return pollBoxModel({
+      element,
+      onBoxModelChange: boxModel => control.changeBoxModel(boxModel),
+    })
+  }, [element, control])
```

Comment:
This could be a `register` and then the polling could instead be in the controller itself. Just food for thought. Both are fine. 

The question you need to ask yourself is "What do I want the API of the RichTextControl to look like?"

Passing an element or passing a boxmodel.